File: src/main/java/com/happystudio/testzilla/model/ProductCategory.java
package com.happystudio.testzilla.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

/**
 * 产品分类的Model.
 * @author Xie Haozhe
 */
@Entity
@Table(name = "tz_product_categories")
public class ProductCategory implements Serializable {
	/**
	 * ProductCategory的默认构造函数.
	 */
	public ProductCategory() {}
	
	/**
	 * ProductCategory的构造函数.
	 * @param productCategoryId - 产品分类的唯一标识符
	 * @param productCategorySlug - 产品分类的唯一英文简称
	 * @param productCategoryName - 产品分类的名称
	 */
	public ProductCategory(int productCategoryId, String productCategorySlug, String productCategoryName) {
		this.productCategoryId = productCategoryId;
		this.productCategorySlug = productCategorySlug;
		this.productCategoryName = productCategoryName;
	}
	
	/**
	 * 获取产品分类的唯一标识符.
	 * @return 产品分类的唯一标识符
	 */
	public int getProductCategoryId() {
		return productCategoryId;
	}

	/**
	 * 设置产品分类的唯一标识符.
	 * @param productCategoryId - 产品分类的唯一标识符
	 */
	public void setProductCategoryId(int productCategoryId) {
		this.productCategoryId = productCategoryId;
	}

	/**
	 * 获取产品分类的唯一英文简称.
	 * @return 产品分类的唯一英文简称
	 */
	public String getProductCategorySlug() {
		return productCategorySlug;
	}

	/**
	 * 设置产品分类的唯一英文简称.
	 * @param productCategorySlug - 产品分类的唯一英文简称
	 */
	public void setProductCategorySlug(String productCategorySlug) {
		this.productCategorySlug = productCategorySlug;
	}

	/**
	 * 获取产品分类的名称.
	 * @return 产品分类的名称
	 */
	public String getProductCategoryName() {
		return productCategoryName;
	}

	/**
	 * 设置产品分类的名称.
	 * @param productCategoryName - 产品分类的名称
	 */
	public void setProductCategoryName(String productCategoryName) {
		this.productCategoryName = productCategoryName;
	}
	
	/* (non-Javadoc)
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		return String.format("ProductCategory [ID=%d, Slug=%s, Name=%s]",
					new Object[] { productCategoryId, productCategorySlug, productCategoryName });
	}

	/**
	 * 产品分类的唯一标识符.
	 */
	@Id
	@GeneratedValue
	@Column(name = "product_category_id")
	private int productCategoryId;
	
	/**
	 * 产品分类的唯一英文简称.
	 */
	@Column(name = "product_category_slug")
	private String productCategorySlug;
	
	/**
	 * 产品分类的名称.
	 */
	@Column(name = "product_category_name")
	private String productCategoryName;
	
	/**
	 * 产品列表(以便1-N关联).
	 */
	@OneToMany(targetEntity = Product.class, 
			fetch = FetchType.LAZY, mappedBy = "productCategory")
	private List<Product> products = new ArrayList<Product>();
	
	/**
	 * 唯一的序列化标识符.
	 */
	private static final long serialVersionUID = -1737227489227837991L;
}

File: app/models/ProductCategory.php
<?php
/**
 * @Author: Haozhe Xie
 * @Date:   2020-01-16 12:54:11
 * @Last Modified by:   Haozhe Xie
 * @Last Modified time: 2020-01-16 13:07:01
 */

use Phalcon\Mvc\Model;

/**
 * The categories of product in the application.
 *
 * @package TestZilla\model\Model\ProductCategory
 */
class ProductCategory extends Model {
    /**
     * Initialize the model.
     * According to the document of Phalcon framework, we have to announce 
     * the 1-N relationship in this function.
     */
    public function initialize() {
        $this->setSource(self::TABLE_NAME);
        $this->belongsTo('product_category_id', 'Product', 'product_category_id');
    }

    /**
     * The getter of the field product_category_id.
     * @return the unique ID of the product category
     */
    public function getProductCategoryId() {
        return $this->product_category_id;
    }

    /**
     * The setter of the field product_category_id.
     * @param  int $productCategoryId - the unique ID of the product category
     */
    public function setProductCategoryId($productCategoryId) {
        $this->product_category_id = $productCategoryId;
    }

    /**
     * The getter of the field product_category_slug.
     * @return the unique slug of the product category.
     */
    public function getProductCategorySlug() {
        return $this->product_category_slug;
    }

    /**
     * The setter of the field product_category_slug.
     * @param String $productCategorySlug - the unique slug of the product category
     */
    public function setProductCategorySlug($productCategorySlug) {
        if ( mb_strlen($productCategorySlug, 'utf-8') > 24 ) {
            throw new InvalidArgumentException('[Model\ProductCategory] The length of productCategorySlug CANNOT exceed 24 characters.');
        }
        $this->product_category_slug = $productCategorySlug;
    }

    /**
     * The getter of the field product_category_name.
     * @return The name of the product category in JSON format
     */
    public function getProductCategoryName() {
        return $this->product_category_name;
    }

    /**
     * The setter of the field product_category_name.
     * @param String $productCategoryName - the name of the product category in JSON format
     */
    public function setProductCategoryName($productCategoryName) {
        $json = json_encode($productCategoryName);
        if ( json_last_error() != JSON_ERROR_NONE ) {
            throw new InvalidArgumentException('[Model\ProductCategory] The productCategoryName seems not a valid JSON.');
        }
        $this->product_category_name = $productCategoryName;
    }

    /**
     * Get the description of the ProductCategory object.
     * @return the description of the ProductCategory object
     */
    public function __toString() {
        return sprintf('ProductCategory: [ID=%d, Slug=%s, Name=%s]', 
                $this->product_category_id, $this->product_category_slug, $this->product_category_name);
    }

    /**
     * The unique ID of the product category.
     * @var int
     */
    protected $product_category_id;
    
    /**
     * The unique slug of the product category.
     * @var String
     */
    protected $product_category_slug;

    /**
     * The name of the product category (JSON format).
     * @var String
     */
    protected $product_category_name;

    /**
     * The table name of the model.
     */
    const TABLE_NAME = 'tz_product_categories';
}